core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/image/BlueprintUpgradeOptionValidator.java

```diff
@@ -43,7 +44,8 @@ private BlueprintValidationResult isEnabledForDefaultBlueprint(boolean lockCompo
             LOGGER.debug("Blueprint marks this GA so upgrade is enabled.");
             result = createResult(true);
         } else {
-            result = createResult(lockComponents ? upgradeOption.isOsUpgradeEnabled() : isRuntimeUpgradeEnabled(upgradeOption, dataHubUpgradeEntitled));
+            result = createResult(lockComponents ? upgradeOption.isOsUpgradeEnabled() :
+                    ResourceStatus.USER_MANAGED == blueprint.getStatus() || isRuntimeUpgradeEnabled(upgradeOption, dataHubUpgradeEntitled));
```

Comment:
Is this logic correct? Where this function is called on line #28, it is only called if the status is default, so the status will never be USER_MANAGED here.

Reply:
thank you for pointing it out @nadamdb will correct it.